trio/_socket.py

```diff
@@ -17,7 +30,19 @@
     from collections.abc import Iterable
     from types import TracebackType
 
-    from typing_extensions import Self
+    from typing_extensions import Buffer, Concatenate, ParamSpec, Self, TypeAlias
+
+    from ._abc import HostnameResolver, SocketFactory
+
+    P = ParamSpec("P")
+
+
+T = TypeVar("T")
+
+# must use old-style typing for TypeAlias
```

Comment:
(more specifically, this is because it's at runtime, correct?)

Reply:
Yep. Could quote the whole thing and use `: TypeAlias`, but that then wouldn't work at runtime if there's unbound typevars present.

Reply:
clarified the comment